tests/datasets/test_digital_typhoon.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import os
+import shutil
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+import pytest
+import torch
+import torch.nn as nn
+from _pytest.fixtures import SubRequest
+from pytest import MonkeyPatch
+
+import torchgeo
+from torchgeo.datasets import DatasetNotFoundError, DigitalTyphoonAnalysis
+
+pytest.importorskip('h5py', minversion='3.6')
+
+
+def download_url(url: str, root: str, *args: str, **kwargs: str) -> None:
+    shutil.copy(url, root)
+
+
+class TestDigitalTyphoon:
+    @pytest.fixture(
+        params=[
+            (3, {'wind': 0}, {'pressure': 1500}),
+            (3, {'pressure': 0}, {'wind': 100}),
+        ]
+    )
+    def dataset(
+        self, monkeypatch: MonkeyPatch, tmp_path: Path, request: SubRequest
+    ) -> DigitalTyphoonAnalysis:
+        sequence_length, min_features, max_features = request.param
+        monkeypatch.setattr(
+            torchgeo.datasets.digital_typhoon, 'download_url', download_url
```

Comment:
See #2232, you can now get rid of this line and the above `download_url` function.